### PR TITLE
Adds cross-validation for time series 

### PR DESCRIPTION
**Pull request by aa25desh**
Cross-validation for time series by implementing  train_test_pairs and tsCV.
Working example can be found over
https://github.com/aa25desh/graph/blob/master/add_tscv_to_MLJbase.ipynb

### CODE REVIEW

File: src/resampling.jl [hunk 1]
```diff
@@ -272,6 +272,59 @@ function train_test_pairs(stratified_cv::StratifiedCV, rows, X, y)
     return zip(trains_per_fold, tests_per_fold) |> collect
 
 end
+# ----------------------------------------------------------------
+# Time series cross-validation (vanilla)
+
+"""
+    cv = tsCV(3)
+
```
Comment (ablaom) [inline src/resampling.jl:280]: Please add a keyword constructor, and show signature with defaults in the doc-string. You can use CV code as a model.

File: src/resampling.jl [hunk 2]
```diff
@@ -272,6 +272,59 @@ function train_test_pairs(stratified_cv::StratifiedCV, rows, X, y)
...
+    train_test_pairs(cv, rows)
+
+Returns an `nfolds`-length iterator of `(train, test)` pairs of
+vectors (row indices), where each `train` and `test` is a sub-vector
+of `rows`.
+
+The nfold observations preceding and following the observation are left out
+in the test set.
+
```
Comment (ablaom) [inline src/resampling.jl:292]: I don't really understand this sentence. `nfolds` is used to count the number of folds, but the way I read this, you are using it to count observations?

**Comment by ablaom** [comment]
@aa25desh Many apologies. I merged your PR prematurely by accident and had to do a revert. (I am doing a bunch of PR's today and got two similar confused 😳 ). Would you please make a completely new PR, as I have had problems trying to be too smart about re-applying reverted PR's. 

Again many apologies.

**Comment by aa25desh** [comment]
No problem:)

**Comment by CameronBieganek** [comment]
@aa25desh You might want to rebase on dev, which incorporates this related PR:

https://github.com/alan-turing-institute/MLJBase.jl/pull/243

The idea in that PR is to spread the remainder of `n_observations / n_folds` evenly among the first `r` folds instead of piling them all into the last fold. More discussion here:

https://github.com/alan-turing-institute/MLJBase.jl/issues/232

I suggest the name `TimeSeriesCV` to be consistent with `CV` and `StratifiedCV`. (@ablaom It doesn't look like nested cross-validation is implemented here.)

**Comment by ablaom** [comment]
@aa25desh Did you still want to make a PR?

**Pull request merged**